### PR TITLE
[CORS-2666] Add presubmit for Nutanix E2E on CCPMSO repository

### PR DESCRIPTION
**Pull request by thunderboltsid**
This will allow us to run Nutanix E2E on demand when needed for CPMS support going forward.

### CODE REVIEW

**Comment by thunderboltsid** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-nutanix-ovn

**Comment by thunderboltsid** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-nutanix-ovn

**Comment by thunderboltsid** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-nutanix-ovn

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml [hunk 1]
```diff
@@ -176,6 +176,21 @@ tests:
...
+  cluster: build01
+  optional: true
+  run_if_changed: pkg/controllers/controlplanemachinesetgenerator/.*|.*nutanix.*
+  steps:
+    cluster_profile: nutanix
+    test:
+    - as: test
+      cli: latest
+      commands: JUNIT_DIR=${ARTIFACT_DIR} make test-e2e
+      from: stable:cluster-api-actuator-pkg
```
Comment (damdo) [inline ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:189]: I think this should be:
```
      commands: |
        make e2e-presubmit
      from: src
```

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml [hunk 2]
```diff
@@ -176,6 +176,21 @@ tests:
   steps:
     cluster_profile: vsphere
     workflow: openshift-e2e-vsphere-etcd-scaling
+- as: e2e-nutanix-ovn
+  cluster: build01
+  optional: true
+  run_if_changed: pkg/controllers/controlplanemachinesetgenerator/.*|.*nutanix.*
```
Comment (damdo) [inline ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:182]: This should be
```
  run_if_changed: pkg/machineproviders/providers/openshift/machine/v1beta1/.*|.*nutanix.*
```

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml [hunk 3]
```diff
@@ -176,6 +176,21 @@ tests:
   steps:
     cluster_profile: vsphere
     workflow: openshift-e2e-vsphere-etcd-scaling
+- as: e2e-nutanix-ovn
+  cluster: build01
```
Comment (damdo) [inline ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:180]: Do we need to specifically run it off of `build01`?

**Comment by damdo** [comment]
@thunderboltsid We have a similar PR for OpenStack which might be useful as an example on what it is normally expected for a CPMS platform integration: https://github.com/openshift/release/pull/38605/files

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml (hunk 3 above)
Comment (thunderboltsid) [reply ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:180]: I think so, that's where we have a VPN for nutanix prism central CI server. @adiantum can you confirm?

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml (hunk 2 above)
Comment (thunderboltsid) [reply ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:182]: Nutanix specific changes didn't happen on that path in that repository, though. https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/200/files

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml (hunk 1 above)
Comment (thunderboltsid) [reply ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:189]: > @thunderboltsid We have a similar PR for OpenStack which might be useful as an example on what it is normally expected for a CPMS platform integration: https://github.com/openshift/release/pull/38605/files

My understanding was that we wanted to create a optional pre-submit that we could invoke when needed. Do we also want a periodic job and a postsubmit job @JoelSpeed ?

**Comment by thunderboltsid** [comment]
> @thunderboltsid We have a similar PR for OpenStack which might be useful as an example on what it is normally expected for a CPMS platform integration: https://github.com/openshift/release/pull/38605/files

My understanding was that we wanted to create a optional pre-submit that we could invoke when needed. Do we also want a periodic job and a postsubmit job @JoelSpeed ?

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml (hunk 1 above)
Comment (thunderboltsid) [reply ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml]: fixed

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml (hunk 3 above)
Comment (adiantum) [reply ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:180]: @damdo In the best of my knowledge VPN operator required to access CI cluster present only on `build01` cluster.

**Comment by JoelSpeed** [comment]
I would continue without the periodic for now, lets get the presubmit working first
/approve

**Comment by damdo** [comment]
SGTM

File: ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml (hunk 3 above)
Comment (damdo) [reply ci-operator/config/openshift/cluster-control-plane-machine-set-operator/openshift-cluster-control-plane-machine-set-operator-main.yaml:180]: Gotcha, thanks for explaining!

**Comment by damdo** [comment]
/pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-nutanix-ovn

**Comment by JoelSpeed** [comment]
> /pj-rehearse pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-nutanix-ovn

This won't work until we've merged the PR right?

**Review by damdo — COMMENTED** [review]
/lgtm

**Comment by damdo** [comment]
Given https://github.com/openshift/release/pull/39933#issuecomment-1573606083
/pj rehearsals-ack

**Comment by damdo** [comment]
Given https://github.com/openshift/release/pull/39933#issuecomment-1573606083

/pj-rehearse ack

**Comment by thunderboltsid** [comment]
> New changes are detected. LGTM label has been removed.

rebased as a build job was failing that was not failing earlier with a expected diff that didn't have anything to do with this job.

**Review by damdo — COMMENTED** [review]
/lgtm

**Pull request merged**